File: practice/medium/0151-Reverse_Words_in_String.py
class Solution:
  def reverseWords(self, s: str) -> str:
    t = s.strip().rstrip() # remove lead/trail whitespace 
    u = list(reversed(t.split(' '))) # split at space and reverse
    v = list(filter(lambda x: x and x != ' ' and x != '', u)) # filter out non-words
    return ' '.join(v) # return list as string

File: practice/csharp/medium/0151_ReverseWordsInString.cs
using System.Linq;

public class Solution {
  public string ReverseWords(string s) {
    // Split string into words at space character
    List<string> words = s.Split(" ").ToList();

    // Remove words that are only space characters
    words = words.Where((string w) => w != "").ToList();

    // Reverse the array
    words.Reverse();
    
    // Nifty: words.ForEach(Console.WriteLine);
    return String.Join(" ", words);
  }
}
